pimd/pim6_cmd.c

```diff
@@ -33,16 +33,445 @@
 
 #include "pimd.h"
 #include "pim6_cmd.h"
+#include "pim_cmd_common.h"
 #include "pim_vty.h"
 #include "lib/northbound_cli.h"
 #include "pim_errors.h"
 #include "pim_nb.h"
+#include "pim_addr.h"
+#include "pim_nht.h"
 
 #ifndef VTYSH_EXTRACT_PL
 #include "pimd/pim6_cmd_clippy.c"
 #endif
 
+DEFPY (show_ipv6_pim_channel,
+       show_ipv6_pim_channel_cmd,
+       "show ipv6 pim [vrf NAME] channel [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM downstream channel info\n"
+       JSON_STR)
+{
+	struct vrf *vrf_struct;
+	bool uj = !!json;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	pim_show_channel(vrf_struct->info, vty, uj);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_ssm_range,
+       show_ipv6_pim_ssm_range_cmd,
+       "show ipv6 pim [vrf NAME] group-type [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM group type\n"
+       JSON_STR)
+{
+	struct vrf *vrf_struct;
+	bool uj = !!json;
+
+	vrf_struct = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!vrf_struct)
+		return CMD_WARNING;
+
+	ip_pim_ssm_show_group_range(vrf_struct->info, vty, uj);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (show_ipv6_pim_interface,
+       show_ipv6_pim_interface_cmd,
+       "show ipv6 pim [mlag$mlag] [vrf NAME] interface [[detail|WORD]$detail] [json$json]",
```

Comment:
Do we need mlag here? Lets remove just the optional mlag keyword here and pass false to the api  pim_show_interfaces_single

Reply:
Fixed, Thankyou